src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -727,6 +736,34 @@ public static Map<String, String> parseTopicToTableMap(String input) {
     return topic2Table;
   }
 
+  /**
+   * Convert a Comma separated key value pairs into a Map
+   *
+   * @param input Provided in KC config
+   * @return Map
+   */
+  public static Map<String, String> parseCommaSeparatedKeyValuePairs(String input) {
+    Map<String, String> pairs = new HashMap<>();
+    boolean isInvalid = false;
+    for (String str : input.split(",")) {
+      String[] tt = str.split(":");
+
+      if (tt.length != 2 || tt[0].trim().isEmpty() || tt[1].trim().isEmpty()) {
+        LOGGER.error(
+            "Invalid {} config format: {}",
+            SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_CLIENT_PARAMETER_OVERRIDE_MAP,
+            input);
+        isInvalid = true;
+        break;
```

Comment:
Isn't this validated above?

Reply:
Agreed, but both the verifications are in separate code path. And I would like to keep both. 
So two code paths are:
1. connector start
2. task start

Task wont start if connector hasnt started but since we cant control KC framework, its good to have double verification, once during creation of connector and once during creation of streaming client.

Reply:
That makes sense - would be great if there was just a single verification step but if its possible for something to fail at a later step for a valid reason we should probably just do it twice